src/useNumberState.ts

```diff
@@ -5,9 +5,9 @@ import { isNumeric } from "./utils";
 export function useNumberState(searchParamName: string, options: { defaultValue: number }) {
   const [searchParams, setSearchParams] = useSearchParams();
 
-  const acquiredSearchParam = searchParams.get(searchParamName);
+  const acquiredSearchParam = searchParams.get(searchParamName); // string | null
 
-  const finalValue = isNumeric(acquiredSearchParam) ? (Number(acquiredSearchParam) as number) : options.defaultValue;
+  const finalValue = acquiredSearchParam ? Number(acquiredSearchParam) : options.defaultValue;
```

Comment:
@abdelfatah-ashour This is not a bug, the original line checks if the value can be converted into a number then it parses it. 

case: the query string is `x=hello`, while we are using `useNumberState`. it will use `Number(hello)` as the value while it's not valid